src/tests/ipc-handlers.test.ts

```diff
@@ -118,3 +123,109 @@ describe( 'isFullscreen', () => {
 		expect( result ).toBe( true );
 	} );
 } );
+
+describe( 'importSite', () => {
+	const mockBackupFile: BackupArchiveInfo = {
+		path: '/path/to/backup.zip',
+		type: 'doo',
+	};
+
+	beforeEach( () => {
+		( importBackup as jest.Mock ).mockReset();
+	} );
+
+	it( 'should throw error if site is not found', async () => {
+		( SiteServer.get as jest.Mock ).mockReturnValue( null );
+
+		await expect(
+			importSite( mockIpcMainInvokeEvent, {
+				id: 'non-existent-id',
+				backupFile: mockBackupFile,
+			} )
+		).rejects.toThrow( 'Site not found.' );
+	} );
+
+	it( 'should import backup successfully', async () => {
+		const mockSite = {
+			details: {
+				id: 'test-site',
+				phpVersion: '8.0',
+			},
+			updateSiteDetails: jest.fn(),
+		};
+		( SiteServer.get as jest.Mock ).mockReturnValue( mockSite );
+		( importBackup as jest.Mock ).mockResolvedValue( {
+			meta: {
+				phpVersion: '8.2',
+			},
+		} );
+
+		const result = await importSite( mockIpcMainInvokeEvent, {
+			id: 'test-site',
+			backupFile: mockBackupFile,
+		} );
+
+		expect( importBackup ).toHaveBeenCalledWith(
+			mockBackupFile,
+			mockSite.details,
+			expect.any( Function ),
+			defaultImporterOptions
+		);
+		expect( mockSite.details.phpVersion ).toBe( '8.2' );
+		expect( result ).toBe( mockSite.details );
+	} );
+
+	it( 'should capture exception in Sentry when import fails', async () => {
+		const mockError = new Error( 'Import failed' );
+		const mockSite = {
+			details: {
+				id: 'test-site',
+			},
+		};
+		( SiteServer.get as jest.Mock ).mockReturnValue( mockSite );
+		( importBackup as jest.Mock ).mockRejectedValue( mockError );
+
+		await expect(
+			importSite( mockIpcMainInvokeEvent, {
+				id: 'test-site',
+				backupFile: mockBackupFile,
+			} )
+		).rejects.toThrow( 'Import failed' );
+
+		expect( Sentry.captureException ).toHaveBeenCalledWith( mockError );
+	} );
+
+	it( 'should send import events to renderer process', async () => {
+		const mockSite = {
+			details: {
+				id: 'test-site',
+			},
+		};
+		const mockParentWindow = {
+			webContents: {
+				send: jest.fn(),
+			},
+			isDestroyed: () => false,
+		};
+		( SiteServer.get as jest.Mock ).mockReturnValue( mockSite );
+		( BrowserWindow.fromWebContents as jest.Mock ).mockReturnValue( mockParentWindow );
+
+		await importSite( mockIpcMainInvokeEvent, {
+			id: 'test-site',
+			backupFile: mockBackupFile,
+		} );
+
+		// Get the onEvent callback that was passed to importBackup
+		const onEvent = ( importBackup as jest.Mock ).mock.calls[ 0 ][ 2 ];
+		const mockEventData = { status: 'importing' };
+
+		// Simulate an import event
+		onEvent( mockEventData );
+
+		expect( mockParentWindow.webContents.send ).toHaveBeenCalledWith(
+			'on-import',
+			mockEventData,
+			'test-site'
+		);
+	} );
```

Comment:
I know we have to mock a lot of stuff in our tests because of Electron, but this test still stuck out to me as very mock-heavy for testing something so simple. I would almost advocate that we remove it… Thoughts, @ashfame?